cpp/src/arrow/util/byte_stream_split.h

```diff
@@ -23,25 +23,33 @@
 #include <stdint.h>
 #include <algorithm>
 
+#if defined(ARROW_HAVE_NEON) || defined(ARROW_HAVE_SSE4_2)
+#include <xsimd/xsimd.hpp>
+// Enable the SIMD for ByteStreamSplit Decoder
+#define ARROW_HAVE_SIMD_DECODE_SPLIT
+#endif
+
 #ifdef ARROW_HAVE_SSE4_2
```

Comment:
Why isn't NEON enabled here?

Reply:
Hi pitrou, thanks for your comments.

I splited `ARROW_HAVE_SIMD_SPLIT` into `ARROW_HAVE_SIMD_DECODE_SPLIT` and `ARROW_HAVE_SIMD_ENCODE_SPLIT` which are used for `ByteStreamSplitDecode` and `ByteStreamSplitEncode` respectively.
This PR is for ByteStreamSplitDecode, so not enabling `Encode ` here for Neon.

How about to enable `Encode` and convert `ByteStreamSplitEncode` to xsimd in another PR and Jira? 
Or, do you prefer putting everything in this single PR? 
Thanks. :)

Reply:
Sorry for the delay. It would be more logical to put everything in this single PR, IMHO.